gemma-rest/src/main/java/ubic/gemma/rest/DatasetsWebService.java

```diff
@@ -981,24 +996,24 @@ public ResponseDataObject<List<ExperimentExpressionLevelsValueObject>> getDatase
         );
     }
 
-    private Response outputDataFile( ExpressionExperiment ee, boolean filter ) throws FilteringException {
+    private Response outputDataFile( ExpressionExperiment ee, boolean filter ) throws FilteringException, IOException {
         ee = expressionExperimentService.thawLite( ee );
         File file = expressionDataFileService.writeOrLocateDataFile( ee, false, filter );
         return this.outputFile( file, DatasetsWebService.ERROR_DATA_FILE_NOT_AVAILABLE, ee.getShortName() );
     }
 
-    private Response outputDesignFile( ExpressionExperiment ee ) {
+    private Response outputDesignFile( ExpressionExperiment ee ) throws IOException {
         ee = expressionExperimentService.thawLite( ee );
         File file = expressionDataFileService.writeOrLocateDesignFile( ee, false );
         return this.outputFile( file, DatasetsWebService.ERROR_DESIGN_FILE_NOT_AVAILABLE, ee.getShortName() );
     }
 
-    private Response outputFile( File file, String error, String shortName ) {
+    private Response outputFile( File file, String error, String shortName ) throws IOException {
         if ( file == null || !file.exists() ) {
             throw new NotFoundException( String.format( error, shortName ) );
         }
         // we remove the .gz extension because we use HTTP Content-Encoding
-        return Response.ok( file )
+        return Response.ok( new GZIPInputStream( new FileInputStream( file ) ) )
```

Comment:
Kind of a hack: we decompress so that it can be recompressed by Jersey's `GzipEncoder`. Unfortunately there's no way around this because compression happens automatically when we add the `Content-Encoding: header`.